lib/rucio/transfertool/fts3_plugins.py

```diff
@@ -0,0 +1,152 @@
+
+        self.transfer_limit = config_get_int(
+            "transfers",
+            option="metadata_byte_limit",
+            raise_exception=False,
+            default=4096,
+        )
+
+        # Use a default if the algorithm isn't supported
+        if not self._supports(self.ALGORITHM_NAME, policy_algorithm):
+            logging.debug(f"Policy Algorithm {policy_algorithm} not found, ignoring.")
+            policy_algorithm = self.DEFAULT
+
+        # If the policy has a supplied and registered init function
+        if self._supports(self.HINTS_NAME, policy_algorithm):
+            self._get_one_algorithm(self.HINTS_NAME, name=policy_algorithm)()
+
+        self.set_in_hints = self._get_one_algorithm(self.ALGORITHM_NAME, name=policy_algorithm)
+
+    @classmethod
+    def register(cls, name: str, func: Callable, init_func: Optional[Callable] = None) -> None:
+        """
+        Register a fts3 transfer plugin
+
+        :param name: name to register under
+        :type name: str
+        :param func: function called by the plugin
+        :type func: Callable
+        :param init_func: Initialization requirements for the plugin, defaults to None
+        :type init_func: Optional[Callable], optional
+        """
+        super()._register(cls.ALGORITHM_NAME, algorithm_dict={name: func})
+        if init_func is not None:
+            super()._register(cls.HINTS_NAME, algorithm_dict={name: init_func})
+
+    def _init_activity_hints(self) -> None:
+        try:
+            self.prority_table = dict(config_get_items("tape_priority"))
+        except NoSectionError:
+            self.prority_table = {}
+
+    def _activity_hints(self, activity_kwargs: dict[str, str], default_prority: str = '20') -> dict[str, dict]:
+        """ Activity Hints - assign a prorioty based on activity"""
+        if "activity" in activity_kwargs:
+            activity = activity_kwargs["activity"].lower()
+
+        else:
+            raise InvalidRequest("`activity` field not found in passed metadata")
+
+        default_prority = self.prority_table.get("default", default_prority)
+        priority = self.prority_table.get(activity, default_prority)
+
+        return {"scheduling_hints": {"priority": priority}}
+
+    def _collocation(self, collocation_func: Callable, hints: dict) -> dict[str, dict]:
+        """
+        Wraps a 'collacation' style plugin for formatting
+
+        :param collocation_func: Function that defines the collocation rules
+        :type collocation_func: Callable
+        :param hints: kwargs utilized by the collocation rules
+        :type hints: dict
+        :return: Collocation hints produced by the collocation_func, wrapped
+        :rtype: dict
+        """
+        return {"collocation_hints": collocation_func(*hints)}
+
+    def _test_collocation(self, *hint: dict) -> dict:
+        return {"0": "", "1": "", "2": "", "3": ""}
+
+    def _default(self, *hints: dict) -> dict:
```

Comment:
if this is meant to be implemented in the child classes, it might be worth using the [`@abstractmethod` decorator](https://docs.python.org/3/library/abc.html#abc.abstractmethod)